autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -217,6 +218,10 @@ def push_output(output_name: str, output_data: Any) -> None:
         credentials, creds_lock = creds_manager.acquire(user_id, credentials_meta.id)
         extra_exec_kwargs[field_name] = credentials
 
+    if node_block.id in AYRSHARE_NODE_IDS:
+        profile_key = creds_manager.store.get_ayrshare_profile_key(user_id)
+        extra_exec_kwargs["profile_key"] = profile_key
```

Comment:
Is profile key considered private or secret? not that it changes how you code this segment

Reply:
Ayrshare says to treat it like a secret, but afaics it's not an authentication key by itself and always has to be used along with an API key. Still, API key + profile key = access to user's socials.